phpcs-scan.php

```diff
@@ -212,6 +212,24 @@ function vipgoci_phpcs_scan_single_file(
 		$file_contents
 	);
 
+	$validation = vipgoci_validate( $temp_file_name, $file_name );
+
+	/*
+	 * Skips the phpcs scan when the validation contains any issue
+	 */
+	if ( 0 !== $validation[ 'total' ] ) {
+		$skipped = array(
+			'file_issues_arr_master'	=> array(),
+			'file_issues_str'		=> array(),
+			'temp_file_name'		=> $temp_file_name,
+			'validation'            => $validation
+		);
+		unlink( $temp_file_name );
+
```

Comment:
In `lint-scan.php` [above](https://github.com/Automattic/vip-go-ci/pull/176/files#diff-8a7186458f366d2d94ecd363f6f23db1b629ffba464ff2821fa3961f7cd57b9fR291) the `vipgoci_set_prs_implicated_skipped_files()` function is called, but not here.

We want the scanning modules (lint, phpcs, svg) to be independent of each other, and end-users being able to enable and disable them as they wish. If linting is disabled, it seems this function is never called. Possibly this function needs to be called here too?

Reply:
That would make sense. However, at this stage, the logic for phpcs-scan is slightly different than the lint-scan. While the calls to create the temp file are done in the prs_implicated iteration for phpcs-scan, the lint one is done per commits; adding that function call would add a few more unnecessary array iterations (for now). 
Also, the project is procedural and, even because of the sequence of execution, even when the lint is disabled, the php-cs will validate the files. So, I wouldn't concern too much about it now.

Reply:
Rewriting my previous comment in other words, what I meant is:

[lint-scan.php calls `vipgoci_set_prs_implicated_skipped_files`](https://github.com/Automattic/vip-go-ci/pull/176/files#diff-8a7186458f366d2d94ecd363f6f23db1b629ffba464ff2821fa3961f7cd57b9fR292)  to set all the implicated files at the end because at that point, the application knows that value; 

[phpcs-scan.php  calls ``vipgoci_set_skipped_file``](https://github.com/Automattic/vip-go-ci/pull/176/files#diff-3c410336e93f47dad45b2700845c92749e35de8c1a8d82b8e8f1d312100ca4e8R590) instead. Because of the way it iterates over the ``$prs_implicated`` (pre-existent logic); 

If we observe the ``vipgoci_set_prs_implicated_skipped_files`` function closely, we can see that it [calls the same  ``vipgoci_set_skipped_file`` function](https://github.com/Automattic/vip-go-ci/pull/176/files#diff-d324959e718f3d5d8ac49fed3363985d839138080b4456e85a82bd2181c7400cR47) that ``phpcs-scan`` calls. 

Having that said, both would work whether one or another is disabled. Do you think that is not enough?

Also, by ``verify that the skipping feature works for PHPCS when PHP Linting is disabled (and vice versa)``, do you mean creating unit tests?

Reply:
> Having that said, both would work whether one or another is disabled. Do you think that is not enough?

My apologies, I misread the logic before. There is nothing that needs to be done here. 

> Also, by verify that the skipping feature works for PHPCS when PHP Linting is disabled (and vice versa), do you mean creating unit tests?

I meant by running `vip-go-ci` with these different combinations of options, just manual testing.